### ISSUE TITLE
Migrating from react

### ISSUE THREAD

**Issue by karaxuna** [developit/preact#904]
I'm trying to migrate to preact. Using typescript. I want to import react like I was importing before: `import * as React from 'react';`. My tsconfig.json looks like this:

    {
        "compilerOptions": {
            "jsx": "react",
            "rootDir": ".",
            "target": "es5",
            "lib": ["es2015", "dom"],
            "module": "commonjs",
            "sourceMap": true,
            "experimentalDecorators": true,
            "typeRoots": [],
            "types": [],
            "baseUrl": "",
            "paths": {
                "react": ["node_modules/preact"],
                "react-dom": ["node_modules/preact"]
            }
        },
        "files": [
            "./src/index.ts"
        ],
        "exclude": [
            "node_modules"
        ]
    }

There is `@types/react` in my project. I want to ignore react types and use `preact.d.ts`. The error I see is:  File `.../dist/preact.d.ts is not a module`.

**Comment by DanielRosenwasser** [comment]
Also @karaxuna, to help your specific problem, you can just let Webpack do the aliasing and let TypeScript continue to think it's using React under the hood.

**Comment by developit** [comment]
Agreed - much better to let webpack do the aliasing, since then it will be applied properly to modules within node_modules that import React.
